### PR TITLE
Add docs page to explain how to create jdaviz-readable products

### PR DESCRIPTION
**Pull request by camipacifici**
### Description

This pull request is to add a documentation page to explain how to create products that can be read into Jdaviz. This is to address a request from the community.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

Under "Other changes": documentation page to explain how to create products that can be read into Jdaviz.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/create_products.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    search = client.find(**options**)
+    results = client.retrieve(search)
+    spec1d = results[0]['Spectrum1D']
+
+    specviz = Specviz()
+    specviz.load_data(spec1d, data_label='my spectrum')
+    specviz.show()
+
+
+Data in fits files
```
Comment (pllim) [inline docs/create_products.rst:34]: FITS is an acronym. Same comment applies to all similar occurrences.
```suggestion
Data in FITS files
```

File: docs/create_products.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Data in a database
+------------------
+
+If the data are stored in a database, we recommend storing a :class:`~specutils.Spectrum1D` object
+per entry. This would allow the user to query the data and visualize it in
+Jdaviz with few lines of code. This is an example with an imaginary database.
+
+.. code-block:: python
+
+    from database import Client
```
Comment (pllim) [inline docs/create_products.rst:21]: This example looks awfully specific. Is this from NOIRLab Sparkle? Maybe better to just link to their doc for this part, and then go from there.

If you must have standalone example, better to use astroquery API that someone else can reproduce without installing NOIRLab specific software.

**Review by pllim — COMMENTED** [review]
Do we really need a new page or should this be a new section in https://github.com/spacetelescope/jdaviz/blob/main/docs/specviz/import_data.rst ? Has this been discussed?

File: docs/create_products.rst (hunk 1 above)
Comment (camipacifici) [reply docs/create_products.rst:34]: thanks, i will correct the rest.

File: docs/create_products.rst (hunk 2 above)
Comment (camipacifici) [reply docs/create_products.rst:21]: I was hoping it would like generic enough. I'll remove the example.

**Comment by camipacifici** [comment]
This is supposed to apply to all data, not just 1D spectra. I do not see it fitting well under Specviz.
Maybe I should make that clearer?

**Comment by pllim** [comment]
> This is supposed to apply to all data

Certainly not images? At least I don't see any mention of images or image cubes.

**Comment by camipacifici** [comment]
You are right, it does not obviously apply to images. I specified at the beginning that it applies to 1D, 2D, and 3D spectroscopic data though.

**Comment by kecnry** [comment]
> Do we really need a new page or should this be a new section in https://github.com/spacetelescope/jdaviz/blob/main/docs/specviz/import_data.rst ?

I think this does make sense as a separate page since (if) it is targeted at developers/contributors more-so than users.  I'd be worried about these many "advanced tangents" living on a main page about loading data, although we could link from there to here.

File: docs/create_products.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
+.. _create_products:
+
+Create Jdaviz-readable Products
+===============================
+
+Spectroscopic data products (1D, 2D, and 3D) can be loaded
+in the different ``jdaviz`` configurations using
+essentially two methods, i.e. loading :class:`~specutils.Spectrum1D` objects, or loading
```
Comment (pllim) [inline docs/create_products.rst:8]:
```suggestion
essentially two methods, i.e., loading :class:`~specutils.Spectrum1D` objects or
```

File: docs/create_products.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+from FITS files. Here, we list a few ways in which data can be packaged to be easily loaded
+into a ``jdaviz`` configuration.
+
+Data in a database
+------------------
+
+If the data are stored in a database, we recommend storing a :class:`~specutils.Spectrum1D` object
+per entry. This would allow the user to query the data and visualize it in
+``jdaviz`` with few lines of code.
```
Comment (pllim) [inline docs/create_products.rst:17]:
```suggestion
``jdaviz`` with few lines of code; also see :ref:`create_product_spectrum1d_obj`.
```

File: docs/create_products.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+Data in FITS files
+------------------
+
+If the data are stored as FITS files, we propose three options: the user can
+adopt any of the formats readable by specutils; or they can create their own
+specialized loader; or they can provide scripts to read the FITS products as
+:class:`~specutils.Spectrum1D` objects.
+
```
Comment (pllim) [inline docs/create_products.rst:26]:
```suggestion

.. _create_product_specutils_loader:

```

File: docs/create_products.rst [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    from specutils import Spectrum1D
+    Spectrum1D.read.list_formats()
+
+The majority are fairly specific to missions and instruments. Four formats
+are more generic and adaptable: ``ASCII``, ``ECSV``, ``tabular-fits``, and
+``wcs1d-fits``. More information on how to create files that are readable by
+these loaders can be found on the `specutils github repository
+<https://github.com/astropy/specutils/tree/main/specutils/io/default_loaders>`_.
+
```
Comment (pllim) [inline docs/create_products.rst:42]:
```suggestion

.. _create_product_dedicated_loader:

```

File: docs/create_products.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+Creating a dedicated loader
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `specutils documentation on how to create a custom loader
+<https://specutils.readthedocs.io/en/stable/custom_loading.html#creating-a-custom-loader>`_
+is available. We are working on the necessary documentation to prompt
+``jdaviz`` to recognize a custom loader developed in ``specutils``.
+
```
Comment (pllim) [inline docs/create_products.rst:50]:
```suggestion

.. _create_product_spectrum1d_obj:

```

File: docs/create_products.rst [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+per entry. This would allow the user to query the data and visualize it in
+``jdaviz`` with few lines of code.
+
+Data in FITS files
+------------------
+
+If the data are stored as FITS files, we propose three options: the user can
+adopt any of the formats readable by specutils; or they can create their own
+specialized loader; or they can provide scripts to read the FITS products as
+:class:`~specutils.Spectrum1D` objects.
```
Comment (pllim) [inline docs/create_products.rst:25]:
```suggestion
If the data are stored as FITS files, we propose three options:

* :ref:`create_product_specutils_loader`
* :ref:`create_product_dedicated_loader`
* :ref:`create_product_spectrum1d_obj`
```

File: docs/create_products.rst [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+.. code-block:: python
+
+    from specutils import Spectrum1D
+    Spectrum1D.read.list_formats()
+
+The majority are fairly specific to missions and instruments. Four formats
+are more generic and adaptable: ``ASCII``, ``ECSV``, ``tabular-fits``, and
+``wcs1d-fits``. More information on how to create files that are readable by
+these loaders can be found on the `specutils github repository
```
Comment (pllim) [inline docs/create_products.rst:40]:
```suggestion
these loaders can be found on the `specutils GitHub repository
```

File: docs/create_products.rst [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+Providing scripts to load the data as Spectrum1D objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If none of the above is an acceptable option, the user can create the data
+products with their custom format and provide scripts or Jupyter Notebooks
+that show how to read the products and create :class:`~specutils.Spectrum1D` objects
+that can be read into ``jdaviz``. More about
+how to create :class:`~specutils.Spectrum1D` objects for the 1D, 2D, and 3D cases can be
+found in the corresponding "Importing data" sections of the various configurations.
```
Comment (pllim) [inline docs/create_products.rst:59]: This order is same as their ordering in Jdaviz sidebar.

```suggestion
found in the corresponding "Importing data" sections of the various configurations:

* :ref:`specviz-import-data`
* :ref:`cubeviz-import-data`
* :ref:`specviz2d-import-data`
* :ref:`mosviz-import-api`
```

**Review by pllim — COMMENTED** [review]
As discussed, please also link to this page from `docs/.../import_data.rst` for Cubeviz, Mosviz, Specviz, and Specviz2D so it is more easily discoverable. You can link using this directive below:

```rst
:ref:`create_products`
```

If you choose to also accept my suggestions via GitHub UI, please do it in batch mode that is available under "Files changed". Alternately, you can also choose to manually apply the suggested changes as you see fit by editing the files directly on your local checkout.

Thank you!

**Comment by camipacifici** [comment]
I accepted all the suggestions (thank you!) and added the reference in the various import_data pages.
Thank you again for the review!

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

File: docs/create_products.rst [hunk 11]
```diff
@@ -0,0 +1,71 @@
+.. _create_products:
+
+Create Jdaviz-readable Products
```
Comment (rosteen) [inline docs/create_products.rst:3]:
```suggestion
Creating Jdaviz-readable Products
```

Comment (rosteen) [reply docs/create_products.rst:3]: This would be more consistent with other section titles.

**Review by rosteen — APPROVED** [review]
LGTM other than one minor suggestion. I'm assuming there won't be any objections to it, so I may just commit it and merge the PR.

File: docs/create_products.rst [hunk 12]
```diff
@@ -0,0 +1,71 @@
+.. _create_products:
+
+Creating Jdaviz-readable Products
+===============================
```
Comment (rosteen) [inline docs/create_products.rst:4]:
```suggestion
=================================
```

**Pull request merged**